app/javascript/components/visualization/controls/ScatterPlotLegend.js

```diff
@@ -84,8 +87,14 @@ function LegendEntry({
 
   /** Toggle state of this legend filter, and accordingly upstream */
   function toggleSelection() {
-    const state = !isShown
-    updateShownTraces(label, state, numLabels)
+    const wasShown = !isShown
+    updateShownTraces(label, wasShown, numLabels)
+
+    const legendEntry = {
+      label, numPoints, numLabels, wasShown, iconColor,
+      hasCorrelations
+    }
+    log('click:scatterlegend:single', legendEntry)
```

Comment:
I'm not sure we need a custom event logging code here--this is what logClickOther() was built for.  You can add the `log-click` class to the legend entries and a data-analytics-name and achieve what you want
  Incidentally, the fact that we have to do either of these approaches seems like a side effect of not using links/buttons for the entries.  Is that an accessibility concern?  Should we change those legend entries to be links or at least have role="button"?

Reply:
I hadn't considered tab-navigability!  I've added `role="button"` as you suggest (94522ee).

Regarding custom event logging: this does two things.  

First, trivially, it eases product analysis by giving [seamless compatibility in Mixpanel for the old event that tracks legend clicks](https://mixpanel.com/s/4E3GG3).  We could adjust the query, of course; or we could just as simply do what's already done here.

Second, more substantively, logging this as a custom event gives us a way to track related properties.  The threshold for what counts as "related" is subjective, but I'd suggest that data that's already at hand (e.g. `numPoints`, `hasCorrelations`) qualifies and has analytics value.